CHANGELOG.md

```diff
@@ -8,6 +8,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+### Added
+
+- Add concurrent-safe `Reset` method to `SpanRecorder` in `go.opentelemetry.io/otel/sdk/trace/tracetest`. (#5994)
```

Comment:
Nit, but I don't know if it's necessary to specify the method is concurrent-safe here. It can be read as "there's another method which isn't concurrent-safe".

Reply:
What if it is adjusted like this?

Add `Reset` method to `SpanRecorder` in `go.opentelemetry.io/otel/sdk/trace/tracetest`.

Reply:
Yes, this would be good.